src/core/infer.ml

```diff
@@ -150,18 +144,13 @@ and check : Ctxt.t -> term -> term -> unit = fun ctx t c ->
       check ctx a Type;
       (* We (hopefully) evaluate [c] to a product, and get its body. *)
       let b =
-        let c = Eval.whnf c in
+        let c = unfold (Eval.whnf c) in
```

Comment:
Isn't the `unfold` useless here?

Reply:
fixed in 225de7aae00d7d62bb41700d48d0a348b7a70034